pkg/volume/csi/csi_attacher.go

```diff
@@ -352,9 +352,8 @@ func (c *csiAttacher) MountDevice(spec *volume.Spec, devicePath string, deviceMo
 
 	if err != nil {
 		glog.Errorf(log("attacher.MountDevice failed: %v", err))
-		if err := removeMountDir(c.plugin, deviceMountPath); err != nil {
-			glog.Error(log("attacher.MountDevice failed to remove mount dir after a NodeStageVolume() error [%s]: %v", deviceMountPath, err))
-			return err
+		if clearErr := removeMountDir(c.plugin, deviceMountPath); clearErr != nil {
+			glog.Error(log("attacher.MountDevice failed to remove mount dir after a NodeStageVolume() error [%s]: %v", deviceMountPath, clearErr))
 		}
 		return err
```

Comment:
Maybe the error returned should include both errors if the removeMountDirErr occurs? Might be too verbose though, open to opinions.

Reply:
Yeah, my first idea for this is to aggregate both error messages too, but this may make the user confused and hard to find the real reason for this failure, so I just return the `NodeStageVolume` error.